src/tests/index.test.js

```diff
@@ -33,4 +33,58 @@ describe('serverless-plugin-cloudfront-distribution-tags', function() {
 
    });
 
+   describe('_addTagsToResource', function() {
+
+      it('adds new tags to an untagged resource', function() {
+         var tags = { key1: 'value1' },
+             resource = { Properties: {} };
+
+         plugin._addTagsToResource(resource, tags);
+
+         expect(resource.Properties.Tags).to.eql([
+            { Key: 'key1', Value: 'value1' },
+         ]);
+      });
+
+      it('adds merges new tags with existing tags on resource', function() {
+         var tags = { key1: 'value1' },
+             resource;
+
+         resource = {
+            Properties: {
+               Tags: [
+                  { Key: 'existing1', Value: 'existingValue1' },
+               ],
+            },
+         };
+
+         plugin._addTagsToResource(resource, tags);
+
+         expect(resource.Properties.Tags).to.eql([
+            { Key: 'key1', Value: 'value1' },
+            { Key: 'existing1', Value: 'existingValue1' },
+         ]);
+      });
+
+      it('prefers the existing tag when new tags contain a tag with the same key', function() {
```

Comment:
please either extend this test or add another test that proves that there will still be a merge happen, e.g.

resource has key1 and key2
stack tags has key1 and key3
result is key1, 2, 3